subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/ImpactAnalysisPolicy.kt

```diff
@@ -1,33 +1,17 @@
 package com.avito.instrumentation.configuration
 
-import com.avito.instrumentation.impact.AnalyzeTestImpactTask
-import com.avito.instrumentation.impact.analyzeTestImpactTask
-import org.gradle.api.Project
-import org.gradle.api.file.RegularFile
-import org.gradle.api.provider.Provider
-import org.gradle.api.tasks.TaskProvider
 import java.io.Serializable
 
-@Suppress("UnstableApiUsage")
 sealed class ImpactAnalysisPolicy : Serializable {
 
     sealed class On : ImpactAnalysisPolicy() {
 
-        abstract fun getArtifact(task: AnalyzeTestImpactTask): Provider<RegularFile>
+        object RunAffectedTests : On()
 
-        fun getTask(project: Project): TaskProvider<AnalyzeTestImpactTask> = project.tasks.analyzeTestImpactTask()
+        // Use RunModifiedTests, it also includes new tests
+        object RunNewTests : On()
```

Comment:
`FindModifiedTestsAction` merges new files with modifies.
Why and how can we have a policy for new tests?

Reply:
i added new RunChangedTests and left new and modified unchanged (uses old impact analysis mechanism) 
let's keep both for some time

Reply:
> and left new and modified unchanged (uses old impact analysis mechanism)

From this code, it's not clear to me what to use on the client side. 
How about adding `@Deprecated` annotation with this information?